[quant][graphmode][refactor] insertObservers for Block

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* #35077 [quant][graphmode] Fold quantized prepacking ops
* #35073 [quant] Make conv2d_prepack and linear_prepack pure
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* #34572 [quant][graphmode] quantization support for aten::add
* #34518 [quant][graphmode] quantization work for prim::If
* #34571 [quant][graphmode][fix] use observed_values_ to check values are observed
* **#34414 [quant][graphmode][refactor] insertObservers for Block**
* #34411 [quant][graphmode][fix] Insert dequantize before use node
* #34349 [quant][graphmode][fix] preserve the type of original value when inserting dequant node

Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

jerryzh168 committed:
[quant][graphmode][refactor] insertObservers for Block

Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -466,6 +466,12 @@ class InsertObserversHelper {
                   bool is_entry_point = false,
                   std::unordered_set<Value*> graph_observed_values = std::unordered_set<Value*>());
 
+  std::tuple<OptionalModuleVector, OptionalModuleVector, std::vector<size_t>> insertObserversFor(
```
Please add a test with if statements to verify the functionality here

jerryzh168 replied:
support for prim::If will come later

jerryzh168 replied:
this is just a refactor, not adding new functionalities.

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f reviewed (approved):
lgtm

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] insertObservers for Block"


Summary:
Previously we insert observers for Graph(graph is a wrapper around a Block),
this PR added insertObservers for Block, so that the code can work for the nodes that have sub blocks.

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519604](https://our.internmc.facebook.com/intern/diff/D20519604)

[ghstack-poisoned]
